src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -600,8 +611,17 @@ private void updateDeploymentResultAsFailed(Deployment deployment, DeploymentTas
         } else {
             process = CompletableFuture.completedFuture(result);
         }
-        currentDeploymentTaskMetadata = new DeploymentTaskMetadata(deployment, deploymentTask, process,
-                new AtomicInteger(1), false);
+        currentDeploymentTaskMetadata =
+                new DeploymentTaskMetadata(deployment, deploymentTask, process, new AtomicInteger(1), false);
+    }
+
+    private void updateStatusDetailsFromException(Map<String, Object> statusDetails, Throwable failureCause,
+                                                  DeploymentType deploymentType) {
+        Pair<List<String>, List<String>> errorReport =
+                DeploymentErrorCodeUtils.generateErrorReportFromExceptionStack(failureCause, deploymentType);
+        statusDetails.put(DEPLOYMENT_ERROR_STACK_KEY, errorReport.getLeft());
+        statusDetails.put(DEPLOYMENT_ERROR_TYPES_KEY, errorReport.getRight());
+        statusDetails.put(DEPLOYMENT_FAILURE_CAUSE_KEY, Utils.generateFailureMessage(failureCause));
```

Comment:
Would it be worth outputting status details to the log here, instead of outputting them subsequent to each invocation of `updateStatusDetailsFromException(..)`?

Reply:
I would prefer to log them with the deployment failure message, so that when customer searches for a single error code in the log, they can find the error code and failure stack trace in the same place.